corehq/apps/users/models.py

```diff
@@ -1216,6 +1214,7 @@ def clear_quickcache_for_user(self):
 
         self.get_by_username.clear(self.__class__, self.username)
         self.get_by_user_id.clear(self.__class__, self.user_id)
+        username_to_user_id.clear(self.username)
```

Comment:
I was thinking about adding users in a bulk restore but I guess it's quite unlikely that a user will try to auth before they've been created.